apps/wallet-mobile/src/features/Discover/BrowserNavigator.tsx

```diff
@@ -13,22 +18,45 @@ const Tab = createStackNavigator<BrowserRoutes>()
 export const BrowserNavigator = () => {
   const {styles} = useStyles()
 
+  const appStorage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const manager = React.useMemo(() => createDappConnector(appStorage, wallet), [appStorage, wallet])
+
   return (
-    <SafeAreaView edges={['left', 'right', 'top']} style={styles.root}>
-      <Tab.Navigator
-        screenOptions={{
-          animationEnabled: false,
-          headerShown: false,
-        }}
-      >
-        <Tab.Screen name="discover-browse-dapp" component={BrowseDappScreen} />
-
-        <Tab.Screen name="discover-search-dapp-in-browser" component={SearchDappInBrowserScreen} />
-      </Tab.Navigator>
-    </SafeAreaView>
+    <DappConnectorProvider manager={manager}>
+      <SafeAreaView edges={['left', 'right', 'top']} style={styles.root}>
+        <Tab.Navigator
+          screenOptions={{
+            animationEnabled: false,
+            headerShown: false,
+          }}
+        >
+          <Tab.Screen name="discover-browse-dapp" component={BrowseDappScreen} />
+
+          <Tab.Screen name="discover-search-dapp-in-browser" component={SearchDappInBrowserScreen} />
+        </Tab.Navigator>
+      </SafeAreaView>
+    </DappConnectorProvider>
   )
 }
 
+const createDappConnector = (appStorage: App.Storage, wallet: YoroiWallet) => {
+  const handlerWallet = {
+    id: wallet.id,
+    networkId: wallet.networkId,
+    confirmConnection: async (origin: string) => {
+      return new Promise<boolean>((resolve) => {
+        Alert.alert('Confirm connection', `Do you want to connect to ${origin}?`, [
```

Comment:
@michaeljscript to leverage translations here, whether this become a hook itself, or pass down strings.